### PR TITLE
doc: correct the csi driver deployment of rancher integration

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
Related issue: https://github.com/harvester/harvester/issues/2755

### CODE REVIEW

**Review by bk201 — APPROVED** [review]
LGTM, thanks!

File: docs/rancher/csi-driver.md [hunk 1]
```diff
@@ -40,65 +40,108 @@ When spinning up a Kubernetes cluster using Rancher RKE2 node driver, the Harves
 
 ![select-harvester-cloud-provider](/img/v1.2/rancher/rke2-cloud-provider.png)
 
-You can also deploy Harvester CSI Driver manually if you have already deployed the Harvester node driver manually. Perform the following steps to manually deploy the Harvester CSI Driver. 
+Suppose you would like to deploy the Harvester CSI driver without the cloud provider manually. You could create the Kubernetes cluster with RKE2 node driver with the default cloud provider (in v2.6, and the previous version is `None`).
```
Comment (guangbochen) [inline docs/rancher/csi-driver.md:43]: is `v2.6` referring to the Rancher version?

File: docs/rancher/csi-driver.md [hunk 2]
```diff
@@ -40,65 +40,108 @@ When spinning up a Kubernetes cluster using Rancher RKE2 node driver, the Harves
...
-```
-# mkdir -p /var/lib/rancher/rke2/etc/config-files
-# vim /var/lib/rancher/rke2/etc/config-files/csi_addon.yaml
-```
-
-1. Install Harvester CSI Driver.
-
-Install `Harvester CSI Driver` from the Rancher marketplace.
-![](/img/v1.2/rancher/install_csi_rancher_marketplace.png)
+- Set up `kubectl` and the kubeconfig file (the kubeconfig should be allow you to access the `Harvester Cluster`):
```
Comment (vickyhella) [inline docs/rancher/csi-driver.md:60]:
```suggestion
- Set up `kubectl` and the kubeconfig file (kubeconfig should allow you to access the `Harvester Cluster`):
```

Comment (vickyhella) [reply docs/rancher/csi-driver.md:60]: not sure if it means `kubeconfig should allow you to access` or `kubeconfig should be allowed to access`.

File: docs/rancher/csi-driver.md [hunk 3]
```diff
@@ -40,65 +40,108 @@ When spinning up a Kubernetes cluster using Rancher RKE2 node driver, the Harves
...
+    ```
+    # export KUBECONFIG=kubeconfig
+    # export PATH="${PATH}:/var/lib/rancher/rke2/bin"
+    ```
+
+#### Deploy Harvester CSI Driver
+
+1. Generate cloud-config
+
+    You can generate the kubeconfig file using the [generate_cloud_provider_config.sh](https://raw.githubusercontent.com/harvester/harvester-csi-driver/master/deploy/generate_cloud_provider_config.sh) script. It is available on the [harvester/harvester-csi-driver](https://github.com/harvester/harvester-csi-driver) repo. You will get the cloud-config and cloud-init data as the following.
```
Comment (vickyhella) [inline docs/rancher/csi-driver.md:71]:
```suggestion
    You can generate the kubeconfig file using the [generate_cloud_provider_config.sh](https://raw.githubusercontent.com/harvester/harvester-csi-driver/master/deploy/generate_cloud_provider_config.sh) script. It is available on the [harvester/harvester-csi-driver](https://github.com/harvester/harvester-csi-driver) repo. You can follow the steps below to get the cloud-config and cloud-init data:
```

File: docs/rancher/csi-driver.md [hunk 4]
```diff
@@ -40,65 +40,108 @@ When spinning up a Kubernetes cluster using Rancher RKE2 node driver, the Harves
...
+    ########## cloud-init user data ############
+    write_files:
+    - encoding: b64
+      content: <content>
+      owner: root:root
+      path: /etc/kubernetes/cloud-config
+      permissions: '0644'
+    ```
+    
+    If you already deploy the **cloud-provider**, you do not need to update the cloud-init user data.
```
Comment (vickyhella) [inline docs/rancher/csi-driver.md:107]:
```suggestion
    If you‘ve already deployed the **cloud-provider**, you do not need to update the cloud-init user data.
```

File: docs/rancher/csi-driver.md [hunk 5]
```diff
@@ -40,65 +40,108 @@ When spinning up a Kubernetes cluster using Rancher RKE2 node driver, the Harves
...
+
+2. Setup cloud-provider-config
```
Comment (vickyhella) [inline docs/rancher/csi-driver.md:109]:
```suggestion
2. Set up cloud-provider-config.
```

File: docs/rancher/csi-driver.md [hunk 6]
```diff
@@ -40,65 +40,108 @@ When spinning up a Kubernetes cluster using Rancher RKE2 node driver, the Harves
...
+
+    Create the config-files directory and add the cloud-provider-config file from the output above (Only need the cloud-config part).
```
Comment (vickyhella) [inline docs/rancher/csi-driver.md:111]:
```suggestion
    Create the config-files directory and add the cloud-provider-config file from the output above (only the cloud-config part).
```

File: docs/rancher/csi-driver.md [hunk 7]
```diff
@@ -40,65 +40,108 @@ When spinning up a Kubernetes cluster using Rancher RKE2 node driver, the Harves
...
 ### Deploying with Harvester K3s Node Driver
 
-- [Generate addon configuration](https://github.com/harvester/harvester-csi-driver/blob/master/deploy/generate_addon_csi.sh) and put it in K3s VMs `/etc/kubernetes/cloud-config`.
+#### Prerequisites
 
-```
-# depend on kubectl to operate the Harvester cluster
-./deploy/generate_addon_csi.sh <serviceaccount name> <namespace>
-```
+The script uses `kubectl` and `jq` to operate the Harvester cluster. The script needs access to the `Harvester Cluster` kubeconfig to work. The `<serviceaccount name>` is usually your guest cluster name, and the `<namespace>` needs to match the namespace of the guest cluster. Referring to the following figure. The cluster name is the `<serviceaccount name>`, and the namespace is the `<namespace>`.
```
Comment (vickyhella) [inline docs/rancher/csi-driver.md:131]:
```suggestion
The script uses `kubectl` and `jq` to operate the Harvester cluster. The script needs access to the `Harvester Cluster` kubeconfig to work. `<serviceaccount name>` is usually your guest cluster name (the value of `Cluster Name` in the figure below), and `<namespace>` needs to match the namespace (the value of `Namespace`) of the guest cluster.
```

File: docs/rancher/csi-driver.md [hunk 8]
```diff
@@ -40,65 +40,108 @@ When spinning up a Kubernetes cluster using Rancher RKE2 node driver, the Harves
...
+![](/img/v1.2/rancher/creating_guest_cluster.png)
 
-- Install `Harvester CSI Driver` from the Rancher marketplace.
+- Install `jq`:
+    ```
+    # apt install jq
+    ```
+
+- Set the kubeconfig file (the kubeconfig should be allow you to access the `Harvester Cluster`):
```
Comment (vickyhella) [inline docs/rancher/csi-driver.md:139]:
```suggestion
- Set the kubeconfig file (the kubeconfig should allow you to access the `Harvester Cluster`):
```

File: docs/rancher/csi-driver.md [hunk 9]
```diff
@@ -40,65 +40,108 @@ When spinning up a Kubernetes cluster using Rancher RKE2 node driver, the Harves
...
+    ```
+    # export KUBECONFIG=kubeconfig
+    ```
 
-  ![](/img/v1.2/rancher/install-harvester-csi-driver-in-k3s.png)
+Then you could follow the steps on the CSI driver deployment of the RKE2 [part](./csi-driver.md#deploy-harvester-csi-driver).
```
Comment (vickyhella) [inline docs/rancher/csi-driver.md:144]:
```suggestion
After this, follow the [Deploy Harvester CSI Driver](./csi-driver.md#deploy-harvester-csi-driver) steps described in the RKE2 section.
```

**Review by vickyhella — COMMENTED** [review]
Just some small suggestions. I only commented on the v1.2 docs, so please also apply the suggestions to the v1.1/v1.0 docs as well. Thanks!

File: docs/rancher/csi-driver.md [hunk 10]
```diff
@@ -40,65 +40,108 @@ When spinning up a Kubernetes cluster using Rancher RKE2 node driver, the Harves
...
-You do not need to change the cloud-config path:
-![](/img/v1.2/rancher/donot_change_cloud_config_path.png)
+    ```
+    # export KUBECONFIG=kubeconfig
+    # export PATH="${PATH}:/var/lib/rancher/rke2/bin"
+    ```
+
+#### Deploy Harvester CSI Driver
+
+1. Generate cloud-config
```
Comment (vickyhella) [inline docs/rancher/csi-driver.md:69]:
```suggestion
1. Generate cloud-config.
```

File: docs/rancher/csi-driver.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Vicente-Cheng) [reply docs/rancher/csi-driver.md]: Hi @vickyhella, @guangbochen.
Yes, I mean the rancher version 2.6.
Thanks for pointing it to clarify.

Comment (Vicente-Cheng) [reply docs/rancher/csi-driver.md]: Yes!
Thanks @guangbochen, @vickyhella for helping it more clear.

**Review by vickyhella — COMMENTED** [review]
LGTM! Please also update the csi-driver docs for v1.0/v1.1 later. Thanks!!

**Comment by Vicente-Cheng** [comment]
> LGTM! Please also update the csi-driver docs for v1.0/v1.1 later. Thanks!!

Thanks @vickyhella!!
Already updated the csi-driver docs for v1.0/v1.1

Also, we need to wait https://github.com/harvester/harvester-csi-driver/pull/24 PR to be merged first.
Once the script naming is changed, the document here needs to change, too.

**Review by vickyhella — APPROVED** [review]

**Comment by Vicente-Cheng** [comment]
Hi @vickyhella, @guangbochen, @bk201, I do the minor update for the script improvement.
Please help to recheck it. If OK, I will squash these two commits. Thanks!

File: docs/rancher/csi-driver.md [hunk 11]
```diff
@@ -143,6 +143,12 @@ The script uses `kubectl` and `jq` to operate the Harvester cluster. The script
 
 After this, follow the [Deploy Harvester CSI Driver](./csi-driver.md#deploy-harvester-csi-driver) steps described in the RKE2 section.
 
+Only the difference is that we need to change the script usage like follows:
```
Comment (vickyhella) [inline docs/rancher/csi-driver.md:146]:
```suggestion
The only difference is that you need to change the script command as follows:
```

File: versioned_docs/version-v1.0/rancher/csi-driver.md [hunk 12]
```diff
@@ -142,4 +142,10 @@ The script uses `kubectl` and `jq` to operate the Harvester cluster. The script
     # export KUBECONFIG=kubeconfig
     ```
 
-After this, follow the [Deploy Harvester CSI Driver](./csi-driver.md#deploy-harvester-csi-driver) steps described in the RKE2 section.
\ No newline at end of file
+After this, follow the [Deploy Harvester CSI Driver](./csi-driver.md#deploy-harvester-csi-driver) steps described in the RKE2 section.
+
+Only the difference is that we need to change the script usage like follows:
```
Comment (vickyhella) [inline versioned_docs/version-v1.0/rancher/csi-driver.md:147]:
```suggestion
The only difference is that you need to change the script command as follows:
```

File: versioned_docs/version-v1.1/rancher/csi-driver.md [hunk 13]
```diff
@@ -143,6 +143,12 @@ The script uses `kubectl` and `jq` to operate the Harvester cluster. The script
 
 After this, follow the [Deploy Harvester CSI Driver](./csi-driver.md#deploy-harvester-csi-driver) steps described in the RKE2 section.
 
+Only the difference is that we need to change the script usage like follows:
```
Comment (vickyhella) [inline versioned_docs/version-v1.1/rancher/csi-driver.md:146]:
```suggestion
The only difference is that you need to change the script command as follows:
```

File: docs/rancher/csi-driver.md [hunk 14]
```diff
@@ -40,66 +40,115 @@ When spinning up a Kubernetes cluster using Rancher RKE2 node driver, the Harves
...
+    ########## cloud-init user data ############
+    write_files:
+    - encoding: b64
+      content: <content>
+      owner: root:root
+      path: /etc/kubernetes/cloud-config
+      permissions: '0644'
+    ```
+    
+    If you‘ve already deployed the **cloud-provider**, you do not need to update the cloud-init user data.
```
Comment (vickyhella) [inline docs/rancher/csi-driver.md:107]:
```suggestion
    If you've already deployed the **cloud-provider**, you do not need to update the cloud-init user data.
```

File: versioned_docs/version-v1.0/rancher/csi-driver.md [hunk 15]
```diff
@@ -41,15 +41,111 @@ When spinning up a Kubernetes cluster using Rancher RKE2 node driver, the Harves
...
+    ########## cloud-init user data ############
+    write_files:
+    - encoding: b64
+      content: <content>
+      owner: root:root
+      path: /etc/kubernetes/cloud-config
+      permissions: '0644'
+    ```
+    
+    If you‘ve already deployed the **cloud-provider**, you do not need to update the cloud-init user data.
```
Comment (vickyhella) [inline versioned_docs/version-v1.0/rancher/csi-driver.md:108]:
```suggestion
    If you've already deployed the **cloud-provider**, you do not need to update the cloud-init user data.
```

File: versioned_docs/version-v1.1/rancher/csi-driver.md [hunk 16]
```diff
@@ -40,65 +40,115 @@ When spinning up a Kubernetes cluster using Rancher RKE2 node driver, the Harves
...
+    ########## cloud-init user data ############
+    write_files:
+    - encoding: b64
+      content: <content>
+      owner: root:root
+      path: /etc/kubernetes/cloud-config
+      permissions: '0644'
+    ```
+    
+    If you‘ve already deployed the **cloud-provider**, you do not need to update the cloud-init user data.
```
Comment (vickyhella) [inline versioned_docs/version-v1.1/rancher/csi-driver.md:107]:
```suggestion
    If you've already deployed the **cloud-provider**, you do not need to update the cloud-init user data.
```

**Review by vickyhella — COMMENTED** [review]
Only some grammar suggestions on a sentence, and also fixed a punctuation mistake that I made earlier. Thanks!

**Comment by Vicente-Cheng** [comment]
> Only some grammar suggestions on a sentence, and also fixed a punctuation mistake that I made earlier. Thanks!

All changes are updated! Thanks!!

File: docs/rancher/csi-driver.md [hunk 17]
```diff
@@ -40,66 +40,115 @@ When spinning up a Kubernetes cluster using Rancher RKE2 node driver, the Harves
...
 ![select-harvester-cloud-provider](/img/v1.2/rancher/rke2-cloud-provider.png)
 
-You can also deploy Harvester CSI Driver manually if you have already deployed the Harvester node driver manually. Perform the following steps to manually deploy the Harvester CSI Driver. 
+If you want to deploy the Harvester CSI driver, but don't want to select the `Harvester` cloud provider, select `Default - RKE2 Embeded` in the `Cloud Provider` field (if you are using Rancher v2.6, select `None`).
 
-1. Generate the addon config.
+![](/img/v1.2/rancher/non-harvester-cloud-provider.png)
 
-:::note
+Then, perform the following steps to deploy the Harvester CSI Driver manually.
```
[the remark this answers is not in the document]
Comment (guangbochen) [reply docs/rancher/csi-driver.md:47]: and this is placed after the Prerequisites

File: docs/rancher/csi-driver.md [hunk 18]
```diff
@@ -40,66 +40,115 @@ When spinning up a Kubernetes cluster using Rancher RKE2 node driver, the Harves
...
+    If you've already deployed the **cloud-provider**, you do not need to update the cloud-init user data.
+
+2. Set up cloud-provider-config.
+
+    Create the config-files directory and add the cloud-provider-config file from the output above (only the cloud-config part).
+
+    ```
+    # mkdir -p /var/lib/rancher/rke2/etc/config-files
+    # vim /var/lib/rancher/rke2/etc/config-files/cloud-provider-config
+    ```
```
Comment (guangbochen) [inline docs/rancher/csi-driver.md:116]: merge with step1 via using the `cloud-init` config.

File: docs/rancher/csi-driver.md [hunk 19]
```diff
@@ -40,66 +40,115 @@ When spinning up a Kubernetes cluster using Rancher RKE2 node driver, the Harves
...
+
+3. Install Harvester CSI Driver.
+
+    Install `Harvester CSI Driver` from the Rancher marketplace. You do not need to change the cloud-config path.
+  ![](/img/v1.2/rancher/install_csi_rancher_marketplace.png)
+
+  ![](/img/v1.2/rancher/donot_change_cloud_config_path.png)
 
-1. Create the addon config file and add the `addon` yaml from the output above.
+The Harvester CSI driver should now be deployed successfully.
```
Comment (guangbochen) [inline docs/rancher/csi-driver.md:125]:
```suggestion
By following the above steps, you should be able to see those CSI driver pods are up and running, and you can verify it by provisioning a new PVC using the default storageClass `harvester.`.
```
image

File: docs/rancher/csi-driver.md [hunk 20]
```diff
@@ -40,66 +40,115 @@ When spinning up a Kubernetes cluster using Rancher RKE2 node driver, the Harves
...
-```
+### Deploying with Harvester K3s Node Driver
 
-1. Install Harvester CSI Driver.
+#### Prerequisites
 
-Install `Harvester CSI Driver` from the Rancher marketplace.
-![](/img/v1.2/rancher/install_csi_rancher_marketplace.png)
+The script uses `kubectl` and `jq` to operate the Harvester cluster. The script needs access to the `Harvester Cluster` kubeconfig to work. `<serviceaccount name>` is usually your guest cluster name (the value of `Cluster Name` in the figure below), and `<namespace>` needs to match the namespace (the value of `Namespace`) of the guest cluster.
+![](/img/v1.2/rancher/creating_guest_cluster.png)
```
Comment (guangbochen) [inline docs/rancher/csi-driver.md:132]: I think u can just link to the RKE2 section instead of repeating the steps.

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Pull request merged**